perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/ResourcesManagerBlImpl.java

```diff
@@ -1079,6 +1081,74 @@ public EnrichedResource filterOnlyAllowedAttributes(PerunSession sess, EnrichedR
 		return assignedGroups;
 	}
 
+	@Override
+	public void activateGroupResourceAssignment(PerunSession sess, Group group, Resource resource, boolean async) throws WrongReferenceAttributeValueException, GroupResourceMismatchException, WrongAttributeValueException, GroupNotDefinedOnResourceException {
+		GroupResourceStatus status = getResourcesManagerImpl().getGroupResourceStatus(sess, group, resource);
+		if (status == GroupResourceStatus.ACTIVE || status == GroupResourceStatus.PROCESSING) {
+			// silently skip
+			return;
+		}
+
+		if (async) {
+			getResourcesManagerImpl().setGroupResourceStatus(sess, group, resource, GroupResourceStatus.PROCESSING);
+			new Thread(() -> {
+				try {
+					getPerunBl().getResourcesManagerBl().processGroupResourceActivation(sess, group, resource);
+				} catch (Exception e) {
+					log.error("Activation of group-resource assignment failed.", e);
+					getResourcesManagerImpl().setGroupResourceStatus(sess, group, resource, GroupResourceStatus.FAILED);
+				}
+			}, "activateGroupResourceAssignment").start();
```

Comment:
This async validation might be more complicated. I think we will need to implement a separate component for it. I would suggest leaving a comment here and in the API, forbid any async calls. We will have to do this for all of the new methods. They won't be completely ready for a release so we should prevent any problems that might happen. Just to be sure, you can also throw here some appropriate runtime exception, e.g. `UnsupportedOperationException`

Reply:
We decided to implement the async validation using Spring's `@Async` execution.